### PR TITLE
Bug#280 fix: Backend: The alias of form view of type should be validated

### PR DESCRIPTION
**Pull request by apghalsasi25**

### CODE REVIEW

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ * @package     TJ-UCM
+ * @subpackage  com_tjucm
+ *
+ * @author      Techjoomla <extensions@techjoomla.com>
+ * @copyright   Copyright (C) 2009 - 2019 Techjoomla. All rights reserved.
+ * @license     http://www.gnu.org/licenses/gpl-2.0.html GNU/GPL
+ */
+// No direct access
+defined('_JEXEC') or die('Restricted access');
+use Joomla\Registry\Registry;
```
Comment (ankush-maherwal) [inline src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php]: @apghalsasi25 remove the unused namespaces from the file

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+					$session->set('updatedTypes', $updatedTypes);
+
+					$result['status']   = '';
+					$result['message']  = "Migration in progress";
+				}
+			}
+
+			// Menus
+			$query->from($db->quoteName('#__menu'));
+			$query->where("link" . "=" . "'index.php?option=com_tjucm&view=itemform'" . "||" . "link" . "=" . "'index.php?option=com_tjucm&view=items'");
```
Comment (ankush-maherwal) [inline src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php]: @apghalsasi25 use Joomla query builder [quoteName()] for column names and [quote()] for values

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	public $description = 'Update UCM Types alias';
+
+	/**
+	 * Subform migration script
+	 *
+	 * @return  void
+	 *
+	 * @since   1.0
+	 */
+	public function migrate()
```
Comment (ankush-maherwal) [inline src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php:32]: @apghalsasi25 The function contain some used and irrelevant code, remove the extra unused lines

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	public $title = "Update Types Alias";
+
+	public $description = 'Update UCM Types alias';
+
+	/**
+	 * Subform migration script
+	 *
+	 * @return  void
+	 *
+	 * @since   1.0
```
Comment (ankush-maherwal) [inline src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php]: @apghalsasi25 check the versions in function comments

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+				{
+					if (in_array($ucmType->id, $updatedTypes))
+					{
+						continue;
+					}
+
+					$ucmTypeTable = JTable::getInstance('Type', 'TjucmTable', array('dbo', $db));
+					$ucmTypeTable->load($ucmType->id);
+
+					$updatedAlias = JFilterOutput::stringURLSafe($ucmTypeTable->alias);
```
Comment (ankush-maherwal) [inline src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php:59]: @apghalsasi25 add comments in code where and when required

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php (hunk 1 above)
Comment (apghalsasi25) [reply src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php]: All unused namespaces have been removed from file.

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php (hunk 2 above)
Comment (apghalsasi25) [reply src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php]: Joomla query builder [quoteName()] for column names and [quote()] for values is used in the query

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php (hunk 4 above)
Comment (apghalsasi25) [reply src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php]: The version is changes to 1.2.4

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php (hunk 5 above)
Comment (apghalsasi25) [reply src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php:59]: Comments  have been added where reuired.

File: src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php (hunk 3 above)
Comment (apghalsasi25) [reply src/components/com_tjucm/administrator/houseKeeping/1.2.4/updateAlias.php:32]: I have removed unused code from the function

**Pull request merged**